From 9317ff7bb52575d79551f995df9166a1efd59db0 Mon Sep 17 00:00:00 2001
From: Arnon Kaufman <akaufman@habana.ai>
Date: Wed, 25 Oct 2023 12:59:45 +0300
Subject: [PATCH] [SW-15774] package: reshape scripts

As a result of splitting the build products, some scripts needs to
be adjusted. Mainly due to file location or naming.

Change-Id: I2783d8c2069639f3f3e09abcd002af3a0af0227b
Signed-off-by: Arnon Kaufman <akaufman@habana.ai>
---

diff --git a/board/habanalabs/common/post-image-fit-global.sh b/board/habanalabs/common/post-image-fit-global.sh
index 7fa938d..d07a7ac 100755
--- a/board/habanalabs/common/post-image-fit-global.sh
+++ b/board/habanalabs/common/post-image-fit-global.sh
@@ -34,8 +34,8 @@
 HL_SEC_KEY_INDEX=`grep HL_SEC_KEY_INDEX $HL_SEC_FILE | sed 's/^.*\(0x.*\)$/\1/'`
 HL_SEC_FW_VERSION=`grep HL_SEC_FW_VERSION $HL_SEC_FILE | awk '{ print $NF }'`
 # set zephyr apps version
-HL_ZEPHYR_MGMT_VERSION=`cat ${BR2_EXTERNAL}/output/${DEV_PREFIX}_arc/build/zephyr-custom/build-mgmt/buildversion-habanalabs`
-HL_ZEPHYR_PID_VERSION=`cat ${BR2_EXTERNAL}/output/${DEV_PREFIX}_arc/build/zephyr-custom/build-pid/buildversion-habanalabs`
+HL_ZEPHYR_MGMT_VERSION=`cat ${BR2_EXTERNAL}/output/${DEV_PREFIX}_arc/build/zephyr-mgmt-custom/buildversion-habanalabs`
+HL_ZEPHYR_PID_VERSION=`cat ${BR2_EXTERNAL}/output/${DEV_PREFIX}_arc/build/zephyr-pid-custom/buildversion-habanalabs`
 
 COMMON_BASE=${BR2_EXTERNAL}/board/habanalabs/common
 UBOOT_ZEPHYR_ITS_FILE=uboot-zephyr-${DEV_NAME}_${HL_TARGET}.its
diff --git a/board/habanalabs/common/post-image-spi-arc_global.sh b/board/habanalabs/common/post-image-spi-arc_global.sh
index 2299ab2..0117fe6 100755
--- a/board/habanalabs/common/post-image-spi-arc_global.sh
+++ b/board/habanalabs/common/post-image-spi-arc_global.sh
@@ -21,9 +21,9 @@
 PIF_ICCM_BINFILE=${PIF_ICCM_BINFILE:-pcie_pif-iccm.bin}
 PIF_DCCM_BINFILE=${PIF_DCCM_BINFILE:-pcie_pif-dccm.bin}
 
-ZEPHYR_PID_DIR=${BR2_EXTERNAL}/output/${DEV_PREFIX}_arc/build/zephyr-custom/build-pid
-ZEPHYR_MGMT_DIR=${BR2_EXTERNAL}/output/${DEV_PREFIX}_arc/build/zephyr-custom/build-mgmt
-ZEPHYR_EVNT_DIR=${BR2_EXTERNAL}/output/${DEV_PREFIX}_arc/build/zephyr-custom/build-event
+ZEPHYR_PID_DIR=${BR2_EXTERNAL}/output/${DEV_PREFIX}_arc/build/zephyr-pid-custom
+ZEPHYR_MGMT_DIR=${BR2_EXTERNAL}/output/${DEV_PREFIX}_arc/build/zephyr-mgmt-custom
+ZEPHYR_EVNT_DIR=${BR2_EXTERNAL}/output/${DEV_PREFIX}_arc/build/zephyr-event-custom
 
 PRIME_IMAGE=0
 SECOND_IMAGE=1
diff --git a/board/habanalabs/gaudi2/post-image-agents.sh b/board/habanalabs/gaudi2/post-image-agents.sh
index a3eeac5..e423f0f 100755
--- a/board/habanalabs/gaudi2/post-image-agents.sh
+++ b/board/habanalabs/gaudi2/post-image-agents.sh
@@ -25,7 +25,7 @@
 AGENT_ARCH="arm64"
 LOADABLES="security\", \"ppboot\", \"preboot\", \"img-ptr"
 
-HL_PREBOOT_VERSION_SECUREBOOT=`cat ${BR2_EXTERNAL}/output/${HL_DEV}_arc/build/zephyr-custom/build-preboot/buildversion-habanalabs`
+HL_PREBOOT_VERSION_SECUREBOOT=`cat ${BR2_EXTERNAL}/output/${HL_DEV}_arc/build/zephyr-preboot-custom/build/buildversion-habanalabs`
 HL_PPBOOT_VERSION_SECUREBOOT=`cat ${BR2_EXTERNAL}/output/${HL_DEV}_arc/build/preboot-custom/buildversion-habanalabs | cut -d ' ' -f1`
 HL_FW_VERSION=`cut -d- -f2- ${BASE_DIR}/build/linux-custom/localversion-habanalabs`
 # Set hl-*-version = "hl-<device>-<HL_VERSION>[-rc]-fw-<fw ver>[-<git data>]"; inside FIT header
diff --git a/board/habanalabs/gaudi3/post-image-fit.sh b/board/habanalabs/gaudi3/post-image-fit.sh
index 974dcb6..f62235f 100644
--- a/board/habanalabs/gaudi3/post-image-fit.sh
+++ b/board/habanalabs/gaudi3/post-image-fit.sh
@@ -59,9 +59,9 @@
 	hl_extract_zephyr_elf_sections
 popd
 
-MGMT_STAT=${BR2_EXTERNAL}/output/${HL_DEV}/build/zephyr-custom/build-mgmt/zephyr/zephyr.stat
-PID_STAT=${BR2_EXTERNAL}/output/${HL_DEV}/build/zephyr-custom/build-pid/zephyr/zephyr.stat
-EVENT_STAT=${BR2_EXTERNAL}/output/${HL_DEV}/build/zephyr-custom/build-event/zephyr/zephyr.stat
+MGMT_STAT=${BR2_EXTERNAL}/output/${HL_DEV}/build/zephyr-mgmt-custom/zephyr/zephyr.stat
+PID_STAT=${BR2_EXTERNAL}/output/${HL_DEV}/build/zephyr-pid-custom/zephyr/zephyr.stat
+EVENT_STAT=${BR2_EXTERNAL}/output/${HL_DEV}/build/zephyr-event-custom/zephyr/zephyr.stat
 
 # Extract the start address from stat file
 ENTRY_ADDRESS=$(cat $MGMT_STAT | grep "Entry point address" | head -1)
diff --git a/board/habanalabs/gaudi3_arc/post-image-agents.sh b/board/habanalabs/gaudi3_arc/post-image-agents.sh
index 11b6705..05fddea 100644
--- a/board/habanalabs/gaudi3_arc/post-image-agents.sh
+++ b/board/habanalabs/gaudi3_arc/post-image-agents.sh
@@ -8,8 +8,8 @@
 source ${BR2_EXTERNAL}/board/habanalabs/${HL_DEV}/create-fake-fuse-image.sh
 
 HL_COMPRESS=none
-HL_PREBOOT_VERSION_SECUREBOOT=`cat ${BR2_EXTERNAL}/output/${HL_DEV}/build/zephyr-custom/build-preboot/buildversion-habanalabs`
-HL_PPBOOT_VERSION_SECUREBOOT=`cat ${BR2_EXTERNAL}/output/${HL_DEV}/build/preboot-custom/buildversion-habanalabs | cut -d ' ' -f1`
+HL_PREBOOT_VERSION_SECUREBOOT=`cat ${BR2_EXTERNAL}/output/${HL_DEV}/build/zephyr-preboot-custom/buildversion-habanalabs`
+HL_PPBOOT_VERSION_SECUREBOOT=`cat ${BR2_EXTERNAL}/output/${HL_DEV}/build/zephyr-preboot-custom/buildversion-habanalabs | cut -d ' ' -f1`
 HL_UBOOT_VERSION=""
 HL_FW_VERSION=""
 GAUDI3_SRAM_ADDR_LO=0xF0000000
@@ -20,7 +20,7 @@
 AGENT_ARCH="arc"
 LOADABLES="ppboot\", \"preboot\", \"img-ptr\", \"fake-fuse-image\", \"pif-iccm\", \"pif-dccm"
 
-BIN=${BR2_EXTERNAL}/output/${HL_DEV}/build/zephyr-custom/build-agent-fw-loader/zephyr/zephyr*.bin
+BIN=${BR2_EXTERNAL}/output/${HL_DEV}/build/zephyr-fwl-custom/zephyr/zephyr*.bin
 GAUDI3_AGENT_BIN=$(ls ${BIN} 2>/dev/null)
 GAUDI3_AGENT_STAT=$(echo $GAUDI3_AGENT_BIN | sed 's/\.bin/.stat/')
 GAUDI3_SPEC_FW_LOADER_ITS=$(ls ${BR2_EXTERNAL}/board/habanalabs/${HL_DEV}/agent-fw_loader.its)
